ingeniamotion/configuration.py

```diff
@@ -931,3 +933,49 @@ def get_fw_version(self, alias: str, subnode: int) -> str:
         if not isinstance(fw_value, str):
             raise TypeError("Firmware value has to be a string")
         return fw_value
+
+    def get_vendor_id(self, alias: str) -> int:
+        """Get the vendor ID of a drive.
+
+        Args:
+            alias: Alias of the drive.
+
+        Returns:
+            Vendor ID.
+
+        Raises:
+            TypeError: If the read vendor ID has the wrong type.
+
+        """
+        vendor_id = self.mc.communication.get_register(self.VENDOR_ID_REGISTER, alias)
+        if not isinstance(vendor_id, int):
+            raise TypeError(
+                f"Wrong {self.VENDOR_ID_REGISTER} value. Expected int, got {type(vendor_id)}"
+            )
+        return vendor_id
+
+    def change_baudrate(self, baud_rate: CAN_BAUDRATE, servo: str = DEFAULT_SERVO) -> None:
+        """Change a CANopen device's baudrate.
+
+        Args:
+            baud_rate: New baud rate value.
+            servo : servo alias to reference it. ``default`` by default.
+
+        Raises:
+            ValueError: If the servo is not a CANopen device.
+
+        """
+        drive = self.mc._get_drive(servo)
+        net = self.mc._get_network(servo)
+        if not isinstance(net, CanopenNetwork):
+            raise ValueError(f"Servo {servo} is not a CANopen device.")
+        vendor_id = self.get_vendor_id(servo)
+        (
+            (prod_code, _),
+            (rev_number, _),
+            _,
+            (serial_number, _),
+        ) = self.get_drive_info_coco_moco(servo)
```

Comment:
Use get_product_code, get_revision_number and get_serial_number functions instead get_drive_info_coco_moco to get the drive info. This change will reduce the drive calls

Reply:
Done